### PR TITLE
Improves the documentation

### PR DESCRIPTION
**Pull request by zrzka**
* Sync the documentation style (README, ...)
* Proper documentation of the public API
* One note, `AsyncReader`/`SyncReader` documentation in the `windows` module is a copy & paste of the documentation from the `unix` module. Both are public. There's a TODO inside to fix/update this.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# Version master
+
+- Internal refactoring (no breaking changes)
+- Removed unsafe `static mut`
+- Documentation update
+
```
Comment (TimonPost) [inline CHANGELOG.md:7]: - Fixed WinAPi tab bug

File: src/input.rs [hunk 2]
```diff
@@ -3,6 +3,8 @@
 
 use crossterm_utils::Result;
 
+// TODO Do not reexport and create a new common AsyncReader structure in the same way as we
```
Comment (TimonPost) [inline src/input.rs]: we should not re-export but we should create an common async reader. I read this like we shouldn't do both.

File: src/input/unix.rs [hunk 3]
```diff
@@ -88,24 +88,76 @@ impl Input for UnixInput {
...
+/// ```
 pub struct AsyncReader {
     event_rx: Receiver<u8>,
     shutdown: Arc<AtomicBool>,
 }
 
 impl AsyncReader {
-    /// Construct a new instance of the `AsyncReader`.
-    /// The reading will immediately start when calling this function.
+    // TODO Should the new() really be public?
```
Comment (TimonPost) [inline src/input/unix.rs:154]: Nope, it can be pub(crate), however, let's move those to input idea issue and discuss it over there.

**Review by TimonPost — APPROVED** [review]
So far so good

File: CHANGELOG.md (hunk 1 above)
Comment (zrzka) [reply CHANGELOG.md:7]: Fixed.

File: src/input.rs (hunk 2 above)
Comment (zrzka) [reply src/input.rs]: Did update the TODO.

**Pull request merged**